_posts/2020-06-23-sourcify-faq.md

```diff
@@ -0,0 +1,195 @@
+
+### What is NatSpec?
+
+[NatSpec](https://solidity.readthedocs.io/en/latest/natspec-format.html) is short for Ethereum Natural Language Specification 
+Format. Solidity contracts can use NatSpec comments to provide rich documentation for functions, return variables and more. 
+The comments are categorized by developer-focused and end-user-focused messages. The end-user-focused messages can be crucial 
+to users since they describe what the contract interaction will do in easy language. It is highly recommended to fully 
+annotate Solidity contracts using NatSpec for all public interfaces. 
+
+NatSpec comments can be displayed to the user at the time of interaction, e.g. by being promted in the wallet interface 
+before signing a transaction.
+
+## Sourcify
+
+### What is Sourcify?
+
+Sourcify is an initiative which aims to make interacting with smart contracts on the blockchain safer and more transparent 
+for users. 
+
+To achieve this goal, Sourcify supports several efforts to foster adoption of open-source source verification, metadata files 
+and NatSpec comments.
+
+At its core, Sourcify currently maintains
+- an [interface](https://verification.komputing.org/) that helps developers to verify metadata and contract source code.
+- a [contract repository](https://contractrepo.komputing.org/) of all verified contracts.
+- a [monitoring & verifier service](https://github.com/ethereum/sourcify) that checks for new blocks on Ethereum blockchains 
+(mainnet and testnets) and tries to verify them automatically.
+
+Besides the technical infrastructure, Sourcify is also a collective movement to bring transparency and awareness to the space. 
+We try to educate and build bridges between development tools, wallets, interfaces and other components which all play an 
+important role in demystifying interaction with smart contracts for the end user and hence making blockchain interactions safer.
+
+### How does the Sourcify monitoring & verifier service work?
+
+The monitoring and verifier component...
+
+* listens for new blocks on blockchains and fetches transactions from them.
+* checks to see if their metadata and source code is published on IPFS or Swarm.
+* recompiles the code.
+* checks that it is the same as the deployed code.
+* permanently stores the source code, ABI and other metadata on IPFS.
+
+### How is the Sourcify verifier different to verifying my contracts on EtherScan?
+
+Before we dive into the differences: EtherScan is doing invaluable work for the ecosystem with their 
+[contract verficiation](https://etherscan.io/verifyContract) service, which we are very grateful for!
+
+However, there are some minor problems with the current setup in the EtherScan verification: EtherScan "flattens" the code 
+which makes it impossible to access the metadata. You have to trust EtherScan with the accuracy of the validation, 
```

Comment:
```suggestion
which makes it impossible to verify the metadata. You have to trust EtherScan with the accuracy of the validation, 
```

Reply:
I don't think we should phrase it like that. Not even etherscan can recompile it to the last byte. We have to trust whoever uploaded the source code that this is the original source code with the correct natspec comments and variable names.

Reply:
Okay, I'll rephrase this accordingly.